src/main/java/seedu/address/logic/commands/MailEventCommand.java

```diff
@@ -48,6 +49,10 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownPersonList = model.getFilteredPersonList();
         List<Event> lastShownEventList = model.getFilteredEventList();
+        //check if index is in range of eventlist
+        if (eventIndex.getZeroBased() >= lastShownEventList.size() || eventIndex.getZeroBased() < 0) {
+            throw new CommandException(INDEX_OUT_OF_BOUNDS_ERROR_MESSAGE);
+        }
```

Comment:
There's already a invalid index error message in `Messages` class to use

The `eventIndex.getZeroBased() < 0` check is not really needed since this is guaranteed in the `Index` constructor, but it's also ok to leave it there

```suggestion
        //check if index is in range of event list
        if (eventIndex.getZeroBased() >= lastShownEventList.size()) {
            throw new CommandException(MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
        }
```

Actually I checked the index out of bounds case when I write this part of code, but I wrongly positioned it at line `58-61`. Could you help delete those lines? Thanks!

Reply:
ah i see ok will delete and change thx!